### PR TITLE
Use blacklist for determining if app is migrated

### PR DESCRIPTION
**Pull request by Davidslv**
This commit does several things:
- Removes config gem
- Removes config initializer that reads the settings.yml file
- Removes settings.yml file
- uses NON_MIGRATED_APPS constant in Artefact object instead

All this changes aim to ease future changes, since migrating apps to the new tagging is a work in progress at the moment.

This NON_MIGRATED_APPS constant should be in sync with [content-tagger](https://github.com/alphagov/content-tagger) - [blacklisted-apps](https://github.com/alphagov/content-tagger/blob/master/config/blacklisted-apps.yml)

And with [Rummanger](https://github.com/alphagov/rummager) - [NON_MIGRATED_APPS](https://github.com/alphagov/rummager/blob/master/lib/indexer/index_documents.rb#L6) constant.

By removing the config gem we were able to see that  TaggingUpdater was stil coupled with the settings.yml file, if we had not removed it this would had reached production with a silent bug that was not catched by the test suite.

Part of:
https://trello.com/c/tZd23gW5/568-allow-tagging-in-content-tagger-for-apps-that-can-t-be-tagged-yet-s

### CODE REVIEW

File: app/models/enhancements/artefact.rb [hunk 1]
```diff
@@ -38,13 +46,14 @@ def allow_section_tag_changes?
   end
 
   def tagging_migrated?
-    return false unless Settings.apps_with_migrated_tagging
-    return true if self.new_record? && self.owning_app.nil? && Settings.apps_with_migrated_tagging.include?('publisher')
-    Settings.apps_with_migrated_tagging.include?(self.owning_app)
+    return false if new_record_without_owning_app?
+
+    ! NON_MIGRATED_APPS.include?(self.owning_app)
```
Comment (MatMoore) [inline app/models/enhancements/artefact.rb]: Should this be `!NON_MIGRATED_APPS` without the space?

File: test/functional/artefacts_controller_test.rb [hunk 2]
```diff
@@ -153,23 +153,13 @@ class ArtefactsControllerTest < ActionController::TestCase
         assert_select "li[class~=active] a[href=/artefacts/new]"
       end
 
-      context 'if publisher is an app for which the taging has NOT been migrated' do
-
-        should 'render the tags partial if publisher is not an app with migrated tagging' do
-          Settings.stubs(:apps_with_migrated_tagging).returns(%w(smartanswers testapp))
+      context "when publisher app has not been migrated" do
```
Comment (MatMoore) [inline test/functional/artefacts_controller_test.rb]: To clarify this change for other reviewers - beforehand we were testing multiple potential values of the "migrated apps" list, and we already had special cases for when publisher is migrated/non migrated, even though we haven't migrated this yet.

We think it is simpler to just test the current behaviour of the app, so NON_MIGRATED_APPS is now a constant and we have removed one of the contradictory tests here.

**Comment by tijmenb** [comment]
The PR title isn't fully covering the changes: perhaps "Use blacklist for determining if app is migrated"?

Also, this PR changes the behaviour of pages that were previously in `apps_without_tagging`. They were included there because pages owned by that app do not have a `content_id` as we're trying to get rid of them.

For pages without a `content_id` we should not be showing the link to content-tagger, which would go nowhere.

**Comment by Davidslv** [comment]
> Also, this PR changes the behaviour of pages that were previously in apps_without_tagging. They were included there because pages owned by that app do not have a content_id as we're trying to get rid of them.

> For pages without a content_id we should not be showing the link to content-tagger, which would go nowhere.

That's true, I will correct that.

**Comment by Davidslv** [comment]
comments have been addressed.

@tijmenb I've re-implemented the list of apps without tagging, because implementing a check on `content_id` would conflict with someone trying to add a new artefact (http://panopticon.dev.gov.uk/artefacts/new) and I wasn't 100% sure about changing that behaviour.

**Pull request merged**